guides/source/routing.md

```diff
@@ -529,23 +544,27 @@ If you wanted to link to just a magazine:
 <%= link_to 'Magazine details', @magazine %>
 ```
 
-For other actions, you just need to insert the action name as the first element of the array:
+For other actions, you need to insert the action name as the first element of the array, for `edit_magazine_ad_path`:
 
 ```erb
 <%= link_to 'Edit Ad', [:edit, @magazine, @ad] %>
 ```
 
 This allows you to treat instances of your models as URLs, and is a key advantage to using the resourceful style.
 
+NOTE: In order to automatically derive paths and URLs from objects such as `[@magazine, @ad]`, Rails uses methods from `ActiveModel::Naming` and `ActiveModel::Conversion` modules. Specifically the `@magazine.model_name.route_key` returns `magazines` and `@magazine.to_param` returns a string representation of the model's `id`. So the generated path may be something like `/magazines/1/ads/42` for the objects `[@magazine, @ad]`.
+
 [ActionView::RoutingUrlFor#url_for]: https://api.rubyonrails.org/classes/ActionView/RoutingUrlFor.html#method-i-url_for
 
 ### Adding More RESTful Actions
 
-You are not limited to the seven routes that RESTful routing creates by default. If you like, you may add additional routes that apply to the collection or individual members of the collection.
+You are not limited to the seven routes that RESTful routing creates by default. You can add additional routes that apply to the collection or individual members of the collection.
+
+The below sections describe adding member routes and collection routes. The term `member` refers to routes acting on a single element, such as `show`, `update`, or `destroy`. The term `collection` refers to routes acting on multiple, or a collection of, elements, such as the `index` route.
```

Comment:
It's great that we explain what a `member` and `collection` route is here, but we use these terms frequently in the sections above, so maybe it would be useful to introduce these definitions before the first instance where it's being used.

Reply:
Added definition when they are first mentioned further up too.